### PR TITLE
Implement new proximityPrecision

### LINKED ISSUE #511: [v1.6] Support `proximityPrecision` setting
*Following this [central issue](https://github.com/meilisearch/integration-guides/issues/294)*

This is related to a newly introduced feature in [Meilisearch v1.6.0](https://github.com/meilisearch/meilisearch/releases/tag/v1.6.0): the proximity-precision setting.

Refer to [docs for more information](https://www.meilisearch.com/docs/reference/api/settings#proximity-precision).

- [ ] Add the new settings: `proximityPrecision` with `get`, `update`, and `reset` methods associated.
Here are the JS equivalents in meilisearch-js you should create for this repository:

```jsx
client.index('indexName').getProximityPrecision(); // calls GET /indexes/:uid/settings/proximity-precision
client.index('indexName').updateProximityPrecision('byAttribute');  // calls PUT /indexes/:uid/settings/proximity-precision
client.index('indexName').resetProximityPrecision(); // calls DELETE /indexes/:uid/settings/proximity-precision
```

- [ ] The methods associated with the `/settings` API route (to get, reset and update the settings globally) must be able to receive in the payload the new `proximityPrecision` field
- [ ] ⚠️ Add integration tests ⚠️
- [ ] Update the `.code-samples.meilisearch.yaml`
    - [ ]  Add `get_proximity_precision_settings_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/cd9598324329d20d23f1d9200c6c537acaf01f3e/.code-samples.meilisearch.yaml#L1150-L1152
    - [ ]  Add `update_proximity_precision_settings_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/cd9598324329d20d23f1d9200c6c537acaf01f3e/.code-samples.meilisearch.yaml#L1153-L1157
    - [ ]  Add `reset_proximity_precision_settings_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/cd9598324329d20d23f1d9200c6c537acaf01f3e/.code-samples.meilisearch.yaml#L1158C1-L1160

### PR DESCRIPTION
**Pull request by andre-m-dev**
# Pull Request

## Related issue
Fixes #511 

## What does this PR do?
- Add proximityPrecision

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by ellnix** [comment]
Thank you for your continued contributions! Looks good to me so far, I will wait for you to add tests and mark it ready for review.

File: spec/meilisearch/index/settings_spec.rb [hunk 1]
```diff
@@ -52,6 +54,7 @@
       expect(settings['pagination'].transform_keys(&:to_sym)).to eq(default_pagination)
       expect(settings['filterableAttributes']).to eq([])
       expect(settings['sortableAttributes']).to eq([])
+      expect(settings['proximityPrecision']).to eq(default_proximityPrecision)
```
Comment (ellnix) [inline spec/meilisearch/index/settings_spec.rb:57]: `default_proximityPrecision` should be snake_cased to `default_proximity_precision`

File: lib/meilisearch/index.rb [hunk 2]
```diff
@@ -514,5 +514,20 @@ def update_non_separator_tokens(non_separator_tokens_attributes)
     def reset_non_separator_tokens
       http_delete("/indexes/#{@uid}/settings/non-separator-tokens")
     end
+
+    ### SETTINGS - PROXIMITY PRECISION
+
+    def proximity_precision
+      http_get("/indexes/#{@uid}/settings/proximity-precision")
+    end
+    alias get_proximity_precision proximity_precision
```
Comment (ellnix) [inline lib/meilisearch/index.rb:523]: I believe these `get_` aliases should be removed. They were requested to be removed in two of my previous PRs: https://github.com/meilisearch/meilisearch-ruby/pull/493#discussion_r1348725988

**Review by ellnix — CHANGES_REQUESTED** [review]
Just a couple small problems, the rest looks good

File: spec/meilisearch/index/settings_spec.rb (hunk 1 above)
Comment (andre-m-dev) [reply spec/meilisearch/index/settings_spec.rb]: Changed

File: lib/meilisearch/index.rb (hunk 2 above)
Comment (andre-m-dev) [reply lib/meilisearch/index.rb]: Changed

**Comment by andre-m-dev** [comment]
@ellnix sorry for the small mistakes

**Review by ellnix — CHANGES_REQUESTED** [review]
We are almost there, do not stress about making mistakes, it's why we have tests :smile:

File: spec/meilisearch/index/settings_spec.rb [hunk 3]
```diff
@@ -99,6 +103,7 @@
       expect(settings['distinctAttribute']).to be_nil
       expect(settings['stopWords']).to be_empty
       expect(settings['synonyms']).to be_empty
+      expect(settings['proximityPrecision']).to eq(default_proximityPrecision)
```
Comment (ellnix) [inline spec/meilisearch/index/settings_spec.rb:106]: Old version of default_proximity_precision is causing the tests to fail

Comment (andre-m-dev) [reply spec/meilisearch/index/settings_spec.rb:106]: I don't know how to rerun the pipeline in GitHub

Comment (ellnix) [reply spec/meilisearch/index/settings_spec.rb:106]: I can rerun it but the problem is not fixed. Are you sure you pushed your local fixes?

Comment (andre-m-dev) [reply spec/meilisearch/index/settings_spec.rb]: Mea culpa :see_no_evil:

**Review by ellnix — APPROVED** [review]
LGTM

**Comment by ellnix** [comment]
bors merge

**Comment by ellnix** [comment]
bors try

**Comment by ellnix** [comment]
bors merge

**Pull request merged**